### PR TITLE
✨ Very early support for Flutter Web through the Datadog Browser SDK

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

This change adds support for multiple features of the Datadog Browser SDK in Flutter Web.  This is very early and experimental support, but getting this support into the main line early will make planning future developments of the plugin easier.

### How?

Most of the integration tests remain the same from Flutter Mobile to Flutter Web, but the testing method (`flutter test` vs. `flutter drive`) meant that I needed to move several files around to support the different methods.

Additionally, a lot of tests are invalid on web, or return different data then their mobile counterparts. You can see them, and any requests to `Platform.isX` wrapped with

```dart
if (kIsWeb) {
}
```

This is because `Platform.isX` doesn't work on Web.

As this evolves, the hope is we'll be able to remove more and more of the `kIsWeb` conditions.

### Review checklist

- [x] This pull request has appropriate unit and / or integration tests 
- [x] This pull request references a Github or JIRA issue

### CI Configuration (optional)
- [x] Run unit tests
- [x] Run integration tests

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -280,6 +280,34 @@ workflows:
         - tests_path_pattern: "integration_test"
         - additional_params: "-d iPhone --dart-define DD_CLIENT_TOKEN=$DD_CLIENT_TOKEN,DD_APPLICATION_ID=$DD_APPLICATION_ID"
 
+  integration_web:
```
Comment (fuzzybinary) [inline bitrise.yml:283]: Note - this step is not run anywhere on CI yet.  We'll add it in when web support becomes more stable.

File: packages/datadog_flutter_plugin/README.md [hunk 2]
```diff
@@ -27,6 +27,19 @@ Your iOS Podfile must have `use_frameworks!` (which is true by default in Flutte
...
+`⚠️ Datadog support for Flutter Web is still in early development`
+
+On Web, add the following to your `index.html` under your `head` tag:
+
+```html
+<script type="text/javascript" src="https://www.datadoghq-browser-agent.com/datadog-logs-v4.js"></script>
+<script type="text/javascript" src="https://www.datadoghq-browser-agent.com/datadog-rum-slim-v4.js"></script>
+```
+
+This loads the CDN delivered Datadog Logging and RUM Browser SDKs. Note that the synchronous CDN delivered version of the Browser SDK is the only version currently supported by the Flutter plugin.
```
Comment (cswatt) [inline packages/datadog_flutter_plugin/README.md:41]:
```suggestion
This loads the CDN-delivered Datadog Logging and RUM Browser SDKs. Note that the synchronous CDN-delivered version of the Browser SDK is the only version supported by the Flutter plugin.
```

**Review by cswatt — COMMENTED** [review]
left one grammar/style suggestion, otherwise good for docs

File: packages/datadog_flutter_plugin/lib/datadog_flutter_plugin_web.dart [hunk 3]
```diff
@@ -10,34 +10,41 @@ import 'package:flutter_web_plugins/flutter_web_plugins.dart';
...
-  }
+  Future<void> setTrackingConsent(TrackingConsent trackingConsent) async {}
 
   @override
-  Future<void> setUserInfo(
-      String? id, String? name, String? email, Map<String, dynamic> extraInfo) {
-    return Future.value();
-  }
+  Future<void> setUserInfo(String? id, String? name, String? email,
+      Map<String, dynamic> extraInfo) async {}
```
Comment (ncreated) [inline packages/datadog_flutter_plugin/lib/datadog_flutter_plugin_web.dart:35]: What about these methods? Is it that browser SDK doesn't define such interface and these cannot be forwarded to `*Platform.instance` or is it no-op for some other reason?

File: packages/datadog_flutter_plugin/lib/src/logs/ddlogs_web.dart [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+  @override
+  Future<void> addAttribute(
+      String loggerHandle, String key, Object value) async {
+    final logger = _activeLoggers[loggerHandle];
+    logger?.addContext(key, valueToJs(value, 'value'));
+  }
+
+  @override
+  Future<void> addTag(String loggerHandle, String tag, [String? value]) async {}
```
Comment (ncreated) [inline packages/datadog_flutter_plugin/lib/src/logs/ddlogs_web.dart:61]: nit: maybe we can print _"not supported"_ warning or sth else to indicate users that it doesn't do anything.

**Review by ncreated — APPROVED** [review]
Solid effort and very insightful 💪. I left non-blocking feedback, LG!

File: packages/datadog_flutter_plugin/lib/datadog_flutter_plugin_web.dart (hunk 3 above)
Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/lib/datadog_flutter_plugin_web.dart:35]: Actually, it's probably a miss. I don't think I have an integration test fro setUserInfo and I probably should....

Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/lib/datadog_flutter_plugin_web.dart:35]: I'll add in a separate PR though.

File: packages/datadog_flutter_plugin/lib/src/logs/ddlogs_web.dart (hunk 4 above)
Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/lib/src/logs/ddlogs_web.dart:61]: I thought about that, but I worry that would get really spammy adding 'not supported' for things that you use on mobile that aren't translated yet for web...

**Review by 0xnm — APPROVED** [review]
Impressive work!

**Pull request merged**